### PR TITLE
ResDiff integration: Switch to Modulus Logging

### PR DESCRIPTION
**Pull request by mnabian**
# Modulus Pull Request

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

**Review by MortezaMardani — APPROVED** [review]

**Review by NickGeneva — APPROVED** [review]

**Comment by mnabian** [comment]
Verified `test.sh` runs fine with these changes

File: modulus/experimental/resdiff/generate.py [hunk 1]
```diff
@@ -582,8 +575,7 @@ def generate_and_save(dataset, sampler, f: nc.Dataset, generate_fn, batch_size):
                 writer.write_input(channel_name, time_index, image_lr2[0, channel_idx])
             
 
-
-def generate(net, seeds, class_idx, max_batch_size, img_lr=None, pretext=None, **sampler_kwargs):
+def generate(net, seeds, class_idx, max_batch_size, img_lr=None, device=torch.device('cuda'), pretext=None, **sampler_kwargs):
```
Comment (Alexey-Kamenev) [inline modulus/experimental/resdiff/generate.py:578]: Why is `device` needed here? It gets overwritten on line 597.

File: modulus/experimental/resdiff/training/training_loop.py [hunk 2]
```diff
@@ -195,11 +197,11 @@ def print0(self, *message):
         from userlib.auto_resume import AutoResume
         AutoResume.init()
     except ImportError:
-        print('AutoResume not imported')
+        raise ImportError('AutoResume not imported')
```
Comment (Alexey-Kamenev) [inline modulus/experimental/resdiff/training/training_loop.py:200]: Note the change in behavior: the old code will print the message and continues, the new one - re-raises the exception. Is this what you expect?

File: modulus/experimental/resdiff/generate.py (hunk 1 above)
Comment (mnabian) [reply modulus/experimental/resdiff/generate.py:578]: Thanks, fixed it.

**Pull request merged**